src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -794,21 +797,17 @@ private uint GetParameterValEscape(ParameterSymbol parameter)
 
         private uint GetParameterRefEscape(ParameterSymbol parameter)
         {
-            if (UseUpdatedEscapeRules)
-            {
-                return parameter.RefKind is RefKind.None || parameter.EffectiveScope != DeclarationScope.Unscoped ? Binder.TopLevelScope : Binder.ExternalScope;
-            }
-            else
+            return parameter switch
             {
-                // byval parameters can escape to method's top level. Others can escape further.
-                // NOTE: "method" here means nearest containing method, lambda or local function.
-                return parameter.RefKind == RefKind.None ? Binder.TopLevelScope : Binder.ExternalScope;
-            }
+                { RefKind: RefKind.None } or { EffectiveScope: not DeclarationScope.Unscoped } => Binder.TopLevelScope,
+                { EffectiveScope: DeclarationScope.Unscoped, Type.IsRefLikeType: true } => Binder.ReturnOnlyScope,
```

Comment:
>EffectiveScope: DeclarationScope.Unscoped [](http://example.com/codeflow?start=18&length=41)

Is this check unnecessary given the `or` above?

Reply:
Yes. The state machine won't test the property again in this path, though.

It feels like it might be good to either:
- add `RefKind: not RefKind.None` to make the case fully self-describing, or
- remove `EffectiveScope` to try and reduce redundancy.

Reply:
I went with removing.